### PR TITLE
ci: Add check for dereferenced symlinks

### LINKED ISSUE #14997: Consider adding a check for broken symlinks
Currently, we have no check that there are no broken symbolic links in the repository. This can over time lead to a number of broken links emerging in the repository as the files they point to may be moved/deleted/renamed. 

See https://github.com/magma/magma/pull/14996 as an example.

It should be possible to create a CI check that flags when a symbolic link is broken in a PR.

See this for implementations: https://unix.stackexchange.com/questions/34248/how-can-i-find-broken-symlinks

### PR DESCRIPTION
**Pull request by MoritzThomasHuebner**
Closes #14997 

## Summary

Uses a simple bash command to look for broken symlinks: https://unix.stackexchange.com/questions/34248/how-can-i-find-broken-symlinks.

I added this to the docs workflow because changes is docs is where we are most likely to break symbolic links.

## Test Plan

- [x] Introduce a broken symlink and see if the new check picks it up (See branch at commit 0ffa78febbfb4ad0bc7f3b3b3e7913d70234b9b2)
image

- [x] Make sure the check passes if there are no broken symlinks (current state of the branch)

### CODE REVIEW

File: .github/workflows/docs-workflow.yml [hunk 1]
```diff
@@ -95,3 +95,21 @@ jobs:
...
+    needs: path_filter
+    if: ${{ needs.path_filter.outputs.should_not_skip == 'true' }}
+    name: Check for de-referenced links
+    runs-on: ubuntu-20.04
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+      - name: Check for de-referenced links
+        run: |
```
Comment (LKreutzer) [inline .github/workflows/docs-workflow.yml:115]: Nit: Can we specify `shell: bash` for clarity?

File: .github/workflows/docs-workflow.yml [hunk 2]
```diff
@@ -95,3 +95,21 @@ jobs:
         run: |
           git status
           git diff-index --quiet HEAD
+
+  # Fail if there are de-referenced links in the repository
+  check-de-referenced-links:
+    needs: path_filter
+    if: ${{ needs.path_filter.outputs.should_not_skip == 'true' }}
+    name: Check for de-referenced links
+    runs-on: ubuntu-20.04
```
Comment (LKreutzer) [inline .github/workflows/docs-workflow.yml:108]: Nit: Can we use 22.04?

File: .github/workflows/docs-workflow.yml [hunk 3]
```diff
@@ -95,3 +95,21 @@ jobs:
         run: |
           git status
           git diff-index --quiet HEAD
+
+  # Fail if there are de-referenced links in the repository
```
Comment (LKreutzer) [inline .github/workflows/docs-workflow.yml]: I would understand "de-reference" to mean replace by the actual file, which is not what is checked here. Can we simply call them "broken symlinks" (here and below)?

File: .github/workflows/docs-workflow.yml [hunk 4]
```diff
@@ -95,3 +95,21 @@ jobs:
...
+    if: ${{ needs.path_filter.outputs.should_not_skip == 'true' }}
+    name: Check for de-referenced links
+    runs-on: ubuntu-20.04
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+      - name: Check for de-referenced links
+        run: |
+          LINKS=$(find "${MAGMA_ROOT}" -xtype l)
```
Comment (LKreutzer) [inline .github/workflows/docs-workflow.yml]: This command finds broken symlinks, can we rename `LINKS` accordingly, since it is not a list of all links?

**Review by LKreutzer — CHANGES_REQUESTED** [review]

File: .github/workflows/docs-workflow.yml (hunk 2 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/docs-workflow.yml:108]: From what I can tell almost all workflows currently run on ubuntu-20.04. Migrating them to ubuntu-22.04 (or ubuntu-latest) could be another PR. Within the scope of this PR I would rather stay consistent with what is already there.

File: .github/workflows/docs-workflow.yml (hunk 4 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/docs-workflow.yml]: Done

File: .github/workflows/docs-workflow.yml (hunk 3 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/docs-workflow.yml]: Done

File: .github/workflows/docs-workflow.yml (hunk 1 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/docs-workflow.yml:115]: Done

**Review by LKreutzer — APPROVED** [review]

**Pull request merged**